sunbeam-python/sunbeam/features/maintenance/utils.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from os import linesep
+from typing import Any
+
+import click
+from rich.console import Console
+from watcherclient import v1 as watcher
+
+from sunbeam.core.common import (
+    Result,
+    ResultType,
+    get_step_message,
+    run_plan,
+)
+from sunbeam.core.deployment import Deployment
+from sunbeam.core.juju import JujuHelper
+from sunbeam.core.questions import ConfirmQuestion, Question
+from sunbeam.provider.local.steps import LocalClusterStatusStep
+from sunbeam.provider.maas.steps import MaasClusterStatusStep
+from sunbeam.steps.cluster_status import ClusterStatusStep
+from sunbeam.steps.hypervisor import EnableHypervisorStep
+from sunbeam.steps.maintenance import (
+    MicroCephActionStep,
+    RunWatcherAuditStep,
+)
+
+console = Console()
+LOG = logging.getLogger(__name__)
+
+
+def get_node_status(
+    deployment: Deployment,
+    jhelper: JujuHelper,
+    console: Console,
+    show_hints: bool,
+    node: str,
+) -> dict[str, Any]:
+    cluster_status_step: type[ClusterStatusStep]
+    if deployment.type == "local":
+        cluster_status_step = LocalClusterStatusStep
+    else:
+        cluster_status_step = MaasClusterStatusStep
+
+    results = run_plan([cluster_status_step(deployment, jhelper)], console, show_hints)
+    cluster_status = get_step_message(results, cluster_status_step)
+
+    for _, _node in cluster_status["openstack-machines"].items():
```

Comment:
This should not be hardcoded, but can be committed later not to block this PR. If any other changes is requested, you can modify it too here.

Reply:
More prefer separate PR.